temporal/server.go

```diff
@@ -112,9 +112,14 @@ func (s *Server) Start() error {
 		return err
 	}
 
-	tlsFactory, err := encryption.NewTLSConfigProviderFromConfig(s.so.config.Global.TLS)
-	if err != nil {
-		return fmt.Errorf("TLS provider initialization error : %w", err)
+	var tlsFactory encryption.TLSConfigProvider
+	if s.so.params.TLSConfigProvider != nil {
+		tlsFactory = s.so.params.TLSConfigProvider
+	} else {
+		tlsFactory, err = encryption.NewTLSConfigProviderFromConfig(s.so.config.Global.TLS)
+		if err != nil {
+			return fmt.Errorf("TLS provider initialization error : %w", err)
```

Comment:
nit, not yours, but please remove the space:
```suggestion
			return fmt.Errorf("TLS provider initialization error: %w", err)
```